ci/travis_build_parquet_cpp.sh

```diff
@@ -38,7 +38,7 @@ cmake \
     -GNinja \
     -DCMAKE_BUILD_TYPE=debug \
     -DCMAKE_INSTALL_PREFIX=$ARROW_PYTHON_PARQUET_HOME \
-    -DPARQUET_BOOST_USE_SHARED=off \
+    -DPARQUET_BOOST_USE_SHARED=on \
```

Comment:
what's the rationale for this, the symbol linking issue?

Reply:
`import pyarrow.parquet` was segfaulting, I assumed because we're statically linking boost in the parquet build and dynamically in the arrow build. This only shows up when using the regex library.

Reply:
I see, we should be consistent about which we do across the libraries. Part of why I wish we were building all these libraries in a monorepo setting